runtime/src/accounts.rs

```diff
@@ -318,12 +346,19 @@ impl Accounts {
             Self::get_requested_loaded_accounts_data_size_limit(tx, feature_set)?;
         let mut accumulated_accounts_data_size: usize = 0;
 
+        let instruction_accounts = message
+            .instructions()
+            .iter()
+            .flat_map(|instruction| &instruction.accounts)
+            .unique()
+            .collect::<Vec<&u8>>();
+
         let mut accounts = account_keys
             .iter()
             .enumerate()
             .map(|(i, key)| {
                 let mut account_found = true;
-                let mut account_dep_index = None;
+                let account_dep_index = None;
```

Comment:
There is no need for this to be a variable any more. Constant can be propagated to use site.